tensorflow/tensorflow.bzl

```diff
@@ -396,7 +396,7 @@ def tf_cc_binary(
         srcs = srcs + tf_binary_additional_srcs(),
         deps = deps + tf_binary_dynamic_kernel_deps(kernels) + if_mkl_ml(
             [
-                "//third_party/mkl:intel_binary_blob",
+                clean_dep("//third_party/mkl:intel_binary_blob"),
```

Comment:
We have a C++ unit test (google test based) that depends on TensorFlow C++ as well as core. 
Here's the target dependencies defined in `third_party/ngraph/ngraph_tf.BUILD`:
```
tf_cc_test(
    name = "ngraph_tf_tests",
    size = "small",
    srcs = [
        "test/tf_exec.cpp",
        "test/main.cpp",
    ],
    deps = [
        ":ngraph_tf",
        "@com_google_googletest//:gtest",
        "@org_tensorflow//tensorflow/cc:cc_ops",
        "@org_tensorflow//tensorflow/cc:client_session",
        "@org_tensorflow//tensorflow/core:tensorflow",
...
```
One or more of the TF dependencies are defined as `tf_cc_binary(...)` and the definition of `tf_cc_binary(...)` includes the conditional dependencies of MKL as shown in line 399 above. 
It turns out that without the `clean_dep(...)` or `str(Label(...))` which is equivalent, this dependency is resolved as `@ngraph_tf//third_party/mkl` which is causing the bazel error such as the following:
```
ERROR: /tmp/avijitch/.cache/_bazel_avijitch/7ac9ae669f8ee6504932fa4d1b9f916b/external/ngraph_tf/BUILD.bazel:67:1: no such package '@ngraph_tf//third_party/mkl': BUILD file not found on package path and referenced by '@ngraph_tf//:ngraph_tf_tests'
```

After making this changes, the dependencies are properly expressed and the bazel errors are gone as it's now being correctly resolved as `@org_tensorflow//third_party/mkl:BUILD`.

Our observation is that the ngraph unit test build started to fail since the MKL bazel files changes went in a couple of weeks ago. Comparing the before and after those changes, we noticed that several bazel files properly used the `str(Label(...))` but some didn't (like the ones I added in this PR). 

After making the changes proposed in the PR, the ngraph test build and runs. 

Please let me know if you need more details or have any questions. Thanks  :)  

BTW - I used the following bazel query which was very helpful debugging this:
`bazel query 'buildfiles(deps(@ngraph_tf//:ngraph_tf_tests))' --keep_going`

Reply:
Thanks so much for the detailed explanation! LGTM.